spec/unit/rest/incomingPhoneNumber.spec.js

```diff
@@ -0,0 +1,146 @@
+            'status': 'in-use',
+            'subresource_uris': {
+              'assigned_add_ons': '/2010-04-01/Accounts/ACaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/IncomingPhoneNumbers/PNaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/AssignedAddOns.json'
+            }
+          }
+        ],
+        'next_page_uri': null,
+        'page': 0,
+        'page_size': 50,
+        'previous_page_uri': null,
+        'start': 0,
+        'uri': '/2010-04-01/Accounts/ACaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/IncomingPhoneNumbers.json?FriendlyName=friendly_name&Beta=true&PhoneNumber=%2B19876543210&PageSize=50&Page=0'
+      };
+      holodeck.mock(new Response(200, body));
+      client.api.v2010.accounts('ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
+        .incomingPhoneNumbers.each({ done }, () => null);
+    }
+  );
+  it('should call done when limit in the opts object is reached',
+    function (done) {
+      var body = {
+        'end': 0,
+        'first_page_uri': '/2010-04-01/Accounts/ACaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/IncomingPhoneNumbers.json?FriendlyName=friendly_name&Beta=true&PhoneNumber=%2B19876543210&PageSize=1&Page=0',
+        'incoming_phone_numbers': [
+          {
+            'account_sid': 'ACaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+            'address_requirements': 'none',
+            'address_sid': 'ADaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+            'api_version': '2010-04-01',
+            'beta': null,
+            'capabilities': {
+              'voice': true,
+              'sms': false,
+              'mms': true,
+              'fax': false
+            },
+            'date_created': 'Thu, 30 Jul 2015 23:19:04 +0000',
+            'date_updated': 'Thu, 30 Jul 2015 23:19:04 +0000',
+            'emergency_status': 'Active',
+            'emergency_address_sid': 'ADaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+            'emergency_address_status': 'registered',
+            'friendly_name': '(808) 925-5327',
+            'identity_sid': 'RIaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+            'origin': 'origin',
+            'phone_number': '+18089255327',
+            'sid': 'PNaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+            'sms_application_sid': '',
+            'sms_fallback_method': 'POST',
+            'sms_fallback_url': '',
+            'sms_method': 'POST',
+            'sms_url': '',
+            'status_callback': '',
+            'status_callback_method': 'POST',
+            'trunk_sid': null,
+            'uri': '/2010-04-01/Accounts/ACaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/IncomingPhoneNumbers/PNaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.json',
+            'voice_application_sid': '',
+            'voice_caller_id_lookup': false,
+            'voice_fallback_method': 'POST',
+            'voice_fallback_url': null,
+            'voice_method': 'POST',
+            'voice_url': null,
+            'bundle_sid': 'BUaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+            'voice_receive_mode': 'voice',
+            'status': 'in-use',
+            'subresource_uris': {
```

Comment:
nit: I think all these sample response properties are not needed. Just adds more config to the test.

Reply:
I'll go ahead and remove them to keep it simple. Thanks!

Reply:
Done! Removed all of the unnecessary props from the mocked response body.